routing/pathfind.go

```diff
@@ -766,6 +741,44 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 				continue
 			}
 
+			// Check cache for features of the fromNode.
+			fromFeatures, ok := featureCache[fromNode]
+			if !ok {
+				targetNode, err := g.graph.FetchLightningNode(
+					fromNode,
+				)
+				switch {
+
+				// If the node exists and has features, use
+				// them.
+				case err == nil:
+					// Only use features if they are valid.
+					err = feature.ValidateDeps(
+						targetNode.Features,
+					)
+					if err == nil {
+						fromFeatures = targetNode.Features
+					}
+
+				// If an error other than the node not existing
+				// is hit, abort.
+				case err != channeldb.ErrGraphNodeNotFound:
+					return nil, err
+
+				// Otherwise, we couldn't find a node
+				// announcement, populate a blank feature
+				// vector.
+				default:
+					fromFeatures = lnwire.EmptyFeatureVector()
+				}
+
+				// Update cache.
+				featureCache[fromNode] = fromFeatures
+			}
+			if fromFeatures == nil {
+				continue
```

Comment:
If this happens, should we set an empty feature vector?

Reply:
This happens if the node features in the graph are invalid. In that case we skip the node completely. If they want to route, they need to advertize a valid feature set.

Extracted this closure a little bit, too much nesting that makes it harder to understand.